lib/data-types.js

```diff
@@ -417,6 +417,36 @@ class TIME extends ABSTRACT {
   toSql() {
     return 'TIME';
   }
+  validate(value) {
+    if (!Validator.isDate(String(value)) && !Validator.isDate(`01-01-1970 ${String(value)}`)) {
+      throw new sequelizeErrors.ValidationError(util.format('%j is not a valid date', value));
+    }
+    return true;
+  }
+  _stringify(date) {
+    if (date instanceof Date) {
+      return moment(date).format('HH:mm:ss');
+    } 
+    return date;
+  }
+  _sanitize(value, options) {
+    if ((!options || options && !options.raw) && value instanceof Date ) {
+      return moment(value).format('HH:mm:ss');
+    }
+    return value;
+  }
+  _isChanged(value, originalValue) {
+    if (originalValue && !!value &&
+      (value === originalValue ||
+        value instanceof Date && originalValue instanceof Date && value.getTime() === originalValue.getTime())) {
```

Comment:
a note here - if the date is truncated from the original value this could be inaccurate.

Eg, if the original value was `5:23:53.234` the value from the db would be `5:23:53` so it'd be different. 

Not sure if this is a big issue though.

Reply:
Going through docs for the supported databases only mssql supports the millisecond all other time formats only allow up to the second so doesn't seem like a big issue.

Reply:
dunno, still think its worth checking just in case. Should just be `Math.floor(value.getTime() / 1000) === Math.floor(otherValue.getTime() / 1000)`

Reply:
I have added the checks. So now the milliseconds will be ignored while comparing.

Reply:
@AllAwesome497 soft bump :)